.github/workflows/opendownstream-pr.yml

```diff
@@ -0,0 +1,168 @@
+name: 'Open downstream PRs'
+
+on:
+  pull_request_target:
+    branches:
+      - 'main'
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    steps:
+      - name: 'Checkout Self'
+        uses: actions/checkout@v5
+        # This checks out the code from the PR branch itself
+
+      - name: 'Check for Go file changes'
```

Comment:
I think we can just use [paths](https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#example-including-and-excluding-paths)
```
on:
  pull_request_target:
    paths:
      - '**.go'
      - '!vendor/**'
      - '!**.test.go'
```

This way, the GH action won't even be triggered if a go file isn't changed.